src/access/thread-open-mod-access.js

```diff
@@ -0,0 +1,135 @@
+    const mods = this.capabilities['mod']
+    const member = this.capabilities['member']
+
+    if (op === 'ADD') {
+      // Anyone can add entry if open thread
+      if (!this._members) { return true }
+      // Not open thread, any member or mod can add to thread
+      if (members.includes(entry.identity.id)) { return true }
+      if (mods.includes(entry.identity.id)) { return true }
+    }
+
+    if (op === 'DEL') {
+      const hash = entry.payload.value
+      const delEntry = await entryIPFS.fromMultihash(this._ipfs, hash)
+
+      // An id can delete their own entries
+      if (delEntry.identity.id === entry.identity.id) { return true }
+
+      // Mods can't delete other mods entries
+      if (mods.includes(delEntry.identity.id)) { return false }
+
+      // Mods can delete any other entries
+      if (mods.includes(entry.identity.id)) { return true }
+    }
+
+    return false
+  }
+
+  get capabilities () {
+    // TODO dont do this for every entry, save result, update on db change
+    if (this._db) {
+      let mod = []
+      let member = []
+      Object.entries(this._db.index).forEach(entry => {
+        const capability = entry[1].payload.value.capability
+        const id = entry[1].payload.value.id
+        if (capability === 'mod') mod.push(id)
+        if (capability === 'member') member.push(id)
+      })
+      return {mod, member}
+    }
+    return {}
+  }
+
+  get (capability) {
+    return this.capabilities[capability] || []
+  }
+
+  async close () {
+    await this._db.close()
+  }
+
+  async load (address) {
+    if (this._db) { await this._db.close() }
+    // Force '<address>/_access' naming for the database
+    // TODO should memeber and non member threads create different acess controllers
+    // or how to deal with threads with same name with member and non member (different names?
+    //  address of this access controler probably should just change, then different thread name
+    this._db = await this._orbitdb.feed(ensureAddress(address), {
+      // Use moderator access controller
+      accessController: {
+        type: 'moderator-access',
+      },
+      sync: true
+    })
+
+    //TODO Move somehwere else. but try to add id opening, in case they are first to open this thread
+    await this._db.add(this._db.identity.id)
+
+    this._db.events.on('ready', this._onUpdate.bind(this))
+    this._db.events.on('write', this._onUpdate.bind(this))
+    this._db.events.on('replicated', this._onUpdate.bind(this))
+
+    await this._db.load()
+  }
+
+  async save () {
+    // return the manifest data
+    return {
+      address: this._db.address.toString()
+    }
+  }
+
+  async grant (capability, id) {
+    // TODO limit capabilities
+    // TODO  sanitize key
+    await this._db.add({capability, id})
+  }
+
+  /* Private methods */
+  _onUpdate () {
+    // this.emit('updated')
+    // TODO add back
+  }
+
+  /* Factory */
+  static async create (orbitdb, options = {}) {
```

Comment:
Seems like there are a bunch of things missing here, but I assume you have the entire model figured out?